### PR TITLE
Auth flow and HTTP interceptor

### LINKED ISSUE #63: Auth middleware
- [ ] Auth service
- [x] HTTP interceptor

### PR DESCRIPTION
**Pull request by DrunkenToast**
Close #63

### CODE REVIEW

**Comment by DrunkenToast** [comment]
Ready for review to close #63 
Depends on auth branch in API.

**Review by Dries1234 — APPROVED** [review]
Looks good!

File: src/app/interceptors/auth.interceptor.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    // TODO: ignore everything that isnt to API.
+
+    return next.handle(this.setAuthHeader(request)).pipe(
+      tap({
+        next: (res: HttpEvent<any>) => {
+          console.log('Succesful request => forward response');
+        },
+        error: (err: HttpErrorResponse) => {
+          // TODO: Better way for excluding login? Failed login is also status code 401 -> loop
+          if (err.status == 401 && err.url?.includes('login')) {
```
Comment (vdbe) [inline src/app/interceptors/auth.interceptor.ts:27]: The url should **not** include login

**Review by vdbe — CHANGES_REQUESTED** [review]

**Review by vdbe — APPROVED** [review]

**Comment by DrunkenToast** [comment]
The interceptor is ready and behaves according to this image:
image

This requires you to **always** use the Angular HttpClient, not for example fetch, if you're requesting data from the API.

**Pull request merged**